core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/SdkExecutionAttribute.java

```diff
@@ -152,9 +153,19 @@ public class SdkExecutionAttribute {
         CRC32C, Algorithm.CRC32C
     );
 
+    private static final ImmutableMap<Algorithm, ChecksumAlgorithm> CHECKSUM_ALGORITHM_MAP = ImmutableMap.of(
+        Algorithm.SHA256, SHA256,
+        Algorithm.SHA1, SHA1,
+        Algorithm.CRC32, CRC32,
+        Algorithm.CRC32C, CRC32C
+    );
+
     protected SdkExecutionAttribute() {
     }
 
+    /**
+     * Map from the SelectedAuthScheme and the backing ChecksumSpecs value to a new value for ChecksumSpecs.
+     */
```

Comment:
Right now, with this mapping, updates to the signer property reflect in the `ChecksumSpecs` attribute.  This means that if the `ChecksumAlgorithm` property gets set to null, the `ChecksumSpecs` gets set to null as well. If the property gets set to some algorithm, it gets reflected in `ChecksumSpecs`, but `ChecksumSpecs` also retains whatever other data it had at the time (such as `headerName`, `responseValidationAlgorithms`, etc.)

Reply:
Based on offline discussion, we should keep the bi-directional mapping, but update it so that the header-name also gets mapped.

Reply:
After some more testing, it seems that we can't always re-map the header-name - there are tests that assert that a checksum is placed in a certain header, and it's certainly possible a user could also be relying on this. Ideally, we'd only update the header-name when the signer property gets updated, but there's no way to know that for certain with the current code.